railties/lib/rails/commands/console/console_command.rb

```diff
@@ -12,6 +12,53 @@ def filter_backtrace(bt)
       end
     end
 
+    class IRBConsole
+      def initialize
+        require "irb"
+        require "irb/completion"
+        IRB::WorkSpace.prepend(BacktraceCleaner)
+
+        if !Rails.env.local?
+          # Use env var here so users can override them with env var too
+          ENV["IRB_USE_AUTOCOMPLETE"] ||= "false"
```

Comment:
Any way we could set this on the IRB config rather than in `ENV`? Not a big deal, but it's best not to mutate ENV if we can.

Reply:
If we set it with `IRB.conf`, users can't override it with `IRB_USE_AUTOCOMPLETE=true` due to IRB's initialization order. This probably won't affect many users, but because I mentioned it was overridable in the original changelog I think it'd be a breaking-change.

Reply:
Can't you set it in `IRB.conf` under a `unless ENV["IRB_USE_AUTOCOMPLETE"]`. Reading the env is fine, it's changing it that I think would be best avoided.

Reply:
That's a good point. But for that change I want to convert related unit tests into integration (application) tests too, which is a bit out of scope for this PR IMO. Do you mind if I open another PR for it?

Reply:
Yeah, it's a pre-existing issue, so I don't mind. I'll merge once you squash.

Reply:
Done 👍

Reply:
Follow up PR: https://github.com/rails/rails/pull/50814